src/components/DownloadImageDialog/ApplicationInstructions.tsx

```diff
@@ -0,0 +1,211 @@
+import has from 'lodash/has';
+import { interpolateMustache } from './utils';
+import { Box, List, Tab, Tabs, Typography } from '@mui/material';
+import { memo, useEffect, useMemo, useState } from 'react';
+import { MUILinkWithTracking } from '../MUILinkWithTracking';
+import {
+	DeviceType,
+	KeysOfUnion,
+	OsSpecificContractInstructions,
+} from './models';
+import { OrderedListItem } from '../OrderedListItem';
+
+export type OsOptions = ReturnType<typeof getUserOs>;
+
+export const getUserOs = () => {
+	const platform = window.navigator.platform.toLowerCase();
+	if (platform.includes('win')) {
+		return 'Windows';
+	}
+
+	if (platform.includes('mac')) {
+		return 'MacOS';
+	}
+
+	if (platform.includes('x11') || platform.includes('linux')) {
+		return 'Linux';
+	}
+
+	return 'Unknown';
+};
+
+const dtJsonTocontractOsKeyMap = {
+	windows: 'Windows',
+	osx: 'MacOS',
+	linux: 'Linux',
+} as const;
+
+export const ApplicationInstructions = memo(
+	({
+		deviceType,
+		templateData,
+	}: {
+		deviceType: DeviceType;
+		templateData: { dockerImage: string };
+	}) => {
+		const [currentOs, setCurrentOs] = useState<OsOptions>(getUserOs());
+
+		const instructions = useMemo(() => {
+			if (
+				deviceType?.instructions == null ||
+				Array.isArray(deviceType.instructions) ||
+				typeof deviceType.instructions !== 'object'
+			) {
+				return deviceType?.instructions;
```

Comment:
```suggestion
				return deviceType.instructions;
```

Reply:
we need to keep it as if the first rule in the if is true then it would explod : 

if (
		deviceType?.instructions == null ||
		Array.isArray(deviceType.instructions) ||
		typeof deviceType.instructions !== 'object'
	)
	
	deviceType?.instructions == null might be that deviceType is null. We just do that to be sure that we return the exact same type

Reply:
You're right, my bad